ddpui/core/dbt_automation/json_keys_sql.py

```
@@ -0,0 +1,97 @@
+            FROM sample
+            CROSS JOIN LATERAL jsonb_object_keys(
+                CASE
+                    WHEN sample.payload IS NULL THEN '{{}}'::jsonb
+                    WHEN NULLIF(BTRIM(sample.payload::text), '') IS NULL THEN '{{}}'::jsonb
+                    WHEN LEFT(BTRIM(sample.payload::text), 1) <> '{{' THEN '{{}}'::jsonb
+                    WHEN jsonb_typeof(COALESCE(NULLIF(BTRIM(sample.payload::text), '')::jsonb, '{{}}'::jsonb)) = 'object'
+                        THEN COALESCE(NULLIF(BTRIM(sample.payload::text), '')::jsonb, '{{}}'::jsonb)
+                    ELSE '{{}}'::jsonb
+                END
+            ) AS key
+        """
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)